include/linux/blkdev_compat.h

```diff
@@ -311,10 +311,14 @@ bio_set_flags_failfast(struct block_device *bdev, int *flags)
  * the block layers responsibility to choose the correct way to
  * implement these semantics.
  */
-#ifdef WRITE_FLUSH_FUA
+#if defined(WRITE_FLUSH_FUA)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_FLUSH_FUA
-#else
+#elif defined(WRITE_BARRIER)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_BARRIER
+#elif defined(HAVE_REQ_FUA)
+#define	VDEV_WRITE_FLUSH_FUA		REQ_FUA
```

Comment:
So I think we should do `(REQ_SYNC | REQ_NOIDLE | REQ_FUA)` here, since this is used by empty bio, there shouldn't be difference between `REQ_PREFLUSH` or `REQ_FUA`.

As for `REQ_NOIDLE`, the flag is removed in 4.10 since the behaviour is on by default.